### ISSUE TITLE
Logos

### ISSUE THREAD

**Issue by k-yle** [osmlab/name-suggestion-index#6664]
How does NSI/iD decide when to show logos? It seems weird that the flagpole has a logo but the main preset doesn't.

(screenshot is from Microsoft Edge, where logos from FB and twitter are always blocked) 

![image](https://user-images.githubusercontent.com/16009897/173462308-bf9130f7-5087-4284-9013-fc98f60a3858.png)

**Comment by arch0345** [comment]
On nsi.guide it looks like we're using the Facebook logo for McDonald's:
<img width="878" alt="image" src="https://user-images.githubusercontent.com/85302075/173464085-2ac68431-81c5-47d8-9969-3ad959cf82f5.png">

It looks like we currently prefer logos from Facebook/Twitter over ones in the Commons since they're often not square: https://github.com/openstreetmap/iD/issues/6361

**Comment by k-yle** [comment]
thanks @arch0345. I think it would be reasonable to use the logo from Commons as a fallback, if the browser's privacy settings block logos from facebook/twitter, and if the commons logo is squarish. I'm pretty sure we can detect that in javascript.

this is also related to #2641

**Comment by k-yle** [comment]
unfortunately NSI [only exposes one logo URL](https://github.com/osmlab/name-suggestion-index/blob/4de39777/scripts/dist_files.js#L296-L308), not 3 separate URLs for facebook/twitter/commons. This means iD can't fallback to a different URL if the preferred one fails or is blocked :(

I think the only solution is for NSI to expose 2 fields instead of `imageURL`. One field for the "social media logo", and another field for the "wikimedia commons logo". This will increase the size of the index...

Or alternatively, keep `imageURL` as-is, and expose an extra field for the URL from wikimedia commons, if it's different to the `imageURL`.

**Comment by bhousel** [comment]
> unfortunately NSI [only exposes one logo URL](https://github.com/osmlab/name-suggestion-index/blob/4de39777/scripts/dist_files.js#L296-L308), not 3 separate URLs for facebook/twitter/commons. This means iD can't fallback to a different URL if the preferred one fails or is blocked :(

Yes, but this is because the presets we generate need to conform to [iD's preset json schema](https://github.com/ideditor/schema-builder/blob/6b46ed8756532e236e384b0e1db0505ab93c3b48/schemas/preset.json#L62-L65), so there is only one `imageURL` field to put a logo in.

We do include the alternate logos in the `dist/wikidata.json` file ([link](https://github.com/osmlab/name-suggestion-index/blob/main/dist/wikidata.json)) - this is the file that the nsi.guide site consumes to display the alternates - and iD also fetches this file for other reasons, so it's possible for someone to write the code into iD's preset system to fallback to an alternate logo if the user's privacy settings prevent them from loading the preferred logo.
